### ISSUE TITLE
Files not reloaded

### ISSUE THREAD

**Issue by xanadonf** [alcanzar/logstash-filter-augment#5]
Hi Adam, it seems plugin is not maintained any more but we use it intensively.

After a Debian upgrade to buster, it seems updated files are not reloaded any more by the plugin (Logstash/ES : 6.8.3)

If you have any idea of what could be the problem, it would be a great help for us ;o)

**Comment by alcanzar** [comment]
It uses the file modification time to detect changes. If you do an ls-al before and after the file changes, does the time change?

**Comment by xanadonf** [comment]
Yes, it changes after an update or a touch.
 
We're using the plugin for 2 years without any problems. 
After Debian upgrade from Stretch to Buster, it is functional, except this problem... 

After logstash restart, file is loaded :

> sept. 16 11:20:15 recette5 logstash[20219]: [2019-09-16T11:20:15,664][INFO ][logstash.filters.augment ] checking for modified dictionary files
> sept. 16 11:20:15 recette5 logstash[20219]: [2019-09-16T11:20:15,665][INFO ][logstash.filters.augment ] file xxxxx/data/stats/index_catalogs/media_id2encryptedmedia_id.csv has been modified, reloading

Something I am missing probably...

**Comment by xanadonf** [comment]
If that can help, I notice `load_or_refresh_dictionaries` is called (I added @logger instruction). Problem is in the definition.

I did not notice any backward problem in changelog from ruby 2.3 to 2.5.5. 
Neither a possible problem in the latest minor versions of logstash 6.

**Comment by xanadonf** [comment]
.oO( I do not remember I did last commit 2 years ago... :o) )

Question : what is the purpose of these lines ?
```
    if (@next_refresh && @next_refresh + @refresh_interval < Time.now)
      return
    end
```

Once we enter this if statement, we never leave it. And that's my case, I clearly don't understand why it happens suddenly on our staging server.

**Comment by alcanzar** [comment]
It's to rate-limit how often the file is reloaded... there's a refresh_interval paramter to the filter that controls how often you want to check to see if the file is modified

**Comment by xanadonf** [comment]
Imagine there is no refresh at all for some reason.
Time.now is now > @next_refresh && @next_refresh + @refresh_interval.

From now, we stay in this if statement, we never leave it, and files are never refreshed.
That's why I think these lines are dangerous.

Do I miss something?

**Comment by alcanzar** [comment]
Maybe try adding a logging statement in there that logs all of the values?  The logic looks correct to me, so maybe your refresh_interval parameter is set to something really high?

**Comment by xanadonf** [comment]
I did. On a development server without trafic, then no logstash inputs. If no inputs during refresh_interval * 2, I stay in the if statement, which is normal.

I left 60s. And on the contrary I switched to a very low value, 10 seconds... High values should lead to less problems on low logstash activities.

By the way, I saw you develop for translate plugin, which is maintained and favoured. Maybe it's time to switch. Will have a look at it, since I have only single values to translate...
